airbyte-integrations/connectors/source-shopify/source_shopify/config_migrations.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+from typing import Any, List, Mapping
+
+from airbyte_cdk.config_observation import create_connector_config_control_message
+from airbyte_cdk.entrypoint import AirbyteEntrypoint
+from airbyte_cdk.sources import Source
+from airbyte_cdk.sources.message import InMemoryMessageRepository, MessageRepository
+
+
+class MigrateConfig:
+    """
+    This class stands for migrating the config at runtime,
+    while providing the backward compatibility when falling back to the previous source version.
+
+    Specifically, starting from `2.0.1`, the `start_date` property should be not (None or `None`):
+        > "start_date": "2020-01-01"
+    instead of, in `2.0.0` for some older configs, when the `start_date` was not required:
+        > {...}
+    """
+
+    message_repository: MessageRepository = InMemoryMessageRepository()
+    migrate_key: str = "start_date"
+    # default spec value for the `start_date` is `2020-01-01`
+    default_start_date_value: str = "2020-01-01"
+
+    @classmethod
+    def should_migrate(cls, config: Mapping[str, Any]) -> bool:
+        """
+        This method determines whether the config should be migrated to have the new structure with `start_date`,
+        based on the source spec.
+
+        Returns:
+            > True, if the transformation is necessary
+            > False, otherwise.
+        """
+        # If the config was already migrated, there is no need to do this again.
+        # but if the customer has already switched to the new version,
+        # corrected the old config and switches back to the new version,
+        # we should try to migrate the modified old custom reports.
+        none_values: List[str] = [None, "None"]
```

Comment:
Could this be a tuple? Sorry

Reply:
Sure, but what is wrong having the `List` here?